src/foam/graphics/TreeGraph.js

```diff
@@ -263,33 +267,66 @@
              slot.set(newValue);
              return false;
            }
-           slot.set((14*slot.get() + newValue)/15);
+           slot.set((2*slot.get() + newValue)/3);
            return true;
          }
        ],
 
        listeners: [
-         {
-           name: 'doLayout',
-           isFramed: true,
-           documentation: 'Animate layout until positions stabilize',
-           code: function() {
-             var needsLayout = false;
-             // Scale and translate the view to fit in the available window
-             var gw = this.graph.width-110;
-             var w  = this.maxRight - this.maxLeft + 55;
-             if ( w > gw ) {
-               var scaleX = Math.min(1, gw / w);
-               needsLayout = this.convergeTo(this.scaleX$, scaleX) || needsLayout;
-             }
-
-             var x = (-this.maxLeft+25)/w * gw + 55;
-             needsLayout = this.convergeTo(this.x$, x) || needsLayout;
-             if ( this.layout() || needsLayout ) this.doLayout();
-             this.graph.invalidate();
-           }
-         }
+        {
+          name: 'doLayout',
+          isFramed: true,
+          documentation: 'Animate layout until positions stabilize',
+          code: function() {
+            var needsLayout = false;
+            // Scale and translate the view to fit in the available window
+            var gw = this.graph.width-110;
+            var w  = this.maxRight - this.maxLeft + 55;
+
+            var x = (-this.maxLeft+25)/w * gw + 55;
+            needsLayout = this.convergeTo(this.x$, x) || needsLayout;
+            if ( this.layout() || needsLayout ) this.doLayout();
+            else this.graph.updateCWidth();
+          
+            this.graph.invalidate();
+          }
+        }
        ]
      }
+   ],
+
+   listeners: [
+     {
+       name: 'updateCWidth',
+       isFramed: true,
+       code: function() {
+        const maxes = {
+          maxLeft: 0,
```

Comment:
Wouldn't the maxLeft at least be the width of this box? Same for right? Would this fix the need for the hard-coded 400 elsewhere in the code?

Reply:
By box are you referring to the:
**Canvas**
- can't set it up using `this.canvas.cview.width` or `this.width` because otherwise will go in an infinite loop
- hard coded the initial canvas size in of 2000 

**Canvas Container**
- I set it up as 1614 after removing the hard-coded 400 which is the width of the container and it still cuts off:
<img width="1680" alt="Screen Shot 2019-08-16 at 10 46 35 AM" src="https://user-images.githubusercontent.com/26717171/63176072-3e153680-c013-11e9-9e44-7ed99a1d161f.png">

**Root**
- I set it up as `this.root.width`and it still cuts off:
<img width="1680" alt="Screen Shot 2019-08-16 at 10 50 09 AM" src="https://user-images.githubusercontent.com/26717171/63176364-d0b5d580-c013-11e9-83f2-7806e14ae16e.png">

Reply:
Found a solution, I set width as:
`var width = Math.abs(maxes.maxLeft - maxes.maxRight) + this.nodeWidth + this.padding * 2;`

This is because they always get seem to cut off halfway. So `this.nodeWidth` accounts for the 2 (1/2)nodeWidths on both ends of the tree, and the padding * 2 adds the padding to both ends of the tree.

Managed to get it fit almost perfectly, however there's no padding at the left and all the way at the right the lines start to fade.

<img width="1680" alt="Screen Shot 2019-08-16 at 10 57 27 AM" src="https://user-images.githubusercontent.com/26717171/63176926-d95adb80-c014-11e9-9c2a-db6f2e85e041.png">
<img width="1680" alt="Screen Shot 2019-08-16 at 10 57 20 AM" src="https://user-images.githubusercontent.com/26717171/63176927-d95adb80-c014-11e9-8094-d8ac1af2ad18.png">

So I then **increased the padding factor by 2 to 4.** And now it seems to fit better on both ends:
<img width="1680" alt="Screen Shot 2019-08-16 at 10 58 11 AM" src="https://user-images.githubusercontent.com/26717171/63176962-ea0b5180-c014-11e9-81d0-50a5b0a794b7.png">
<img width="1680" alt="Screen Shot 2019-08-16 at 10 58 06 AM" src="https://user-images.githubusercontent.com/26717171/63176963-ea0b5180-c014-11e9-99d2-f7fce8bb9cde.png">